// Copyright 2021 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wfa.panelmatch;

option java_package = "org.wfanet.panelmatch.common.compression";
option java_multiple_files = true;

// Request to compress data
message CompressRequest {
  repeated bytes uncompressed_data = 1;
  bytes dictionary = 2;
}

// Response with compressed data
message CompressResponse {
  repeated bytes compressed_data = 1;
}

// Request to compress data
message DecompressRequest {
  repeated bytes compressed_data = 1;
  bytes dictionary = 2;
}

// Response with decompressed data
message DecompressResponse {
  repeated bytes decompressed_data = 1;
}
